From: Zhongkun He <hezhongkun.hzk@bytedance.com>
Subject: mm: rename the oldflags and parameter in memalloc_flags_*()
Date: Wed, 18 Jun 2025 15:03:28 +0800

The variable name oldflags can indeed be misleading, because it does not
store the complete original value of flags.  Instead, it records which
flags from the given set are not currently set.  So rename it.

Link: https://lkml.kernel.org/r/20250618070328.2192849-1-hezhongkun.hzk@bytedance.com
Signed-off-by: Zhongkun He <hezhongkun.hzk@bytedance.com>
Cc: Kent Overstreet <kent.overstreet@linux.dev>
Cc: Michal Hocko <mhocko@suse.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/sched/mm.h |   15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

--- a/include/linux/sched/mm.h~mm-rename-the-oldflags-and-parameter-in-memalloc_flags_
+++ a/include/linux/sched/mm.h
@@ -322,21 +322,22 @@ static inline void might_alloc(gfp_t gfp
 }
 
 /**
- * memalloc_flags_save - Add a PF_* flag to current->flags, save old value
+ * memalloc_flags_save - Add a PF_* flag to current->flags, return saved flags mask
  *
  * This allows PF_* flags to be conveniently added, irrespective of current
  * value, and then the old version restored with memalloc_flags_restore().
  */
-static inline unsigned memalloc_flags_save(unsigned flags)
+static inline unsigned int memalloc_flags_save(unsigned int flags_mask)
 {
-	unsigned oldflags = ~current->flags & flags;
-	current->flags |= flags;
-	return oldflags;
+	unsigned int saved_flags_mask = ~current->flags & flags_mask;
+
+	current->flags |= flags_mask;
+	return saved_flags_mask;
 }
 
-static inline void memalloc_flags_restore(unsigned flags)
+static inline void memalloc_flags_restore(unsigned int flags_mask)
 {
-	current->flags &= ~flags;
+	current->flags &= ~flags_mask;
 }
 
 /**
_
